src/caf/distribute/gravity_model/multi_area.py

```diff
@@ -0,0 +1,695 @@
+        if len(row_targets) != cost_matrix.shape[0]:
+            raise IndexError("row_targets doesn't match cost_matrix")
+        if len(col_targets) != cost_matrix.shape[1]:
+            raise IndexError("col_targets doesn't match cost_matrix")
+        if params is not None:
+            self.tlds = pd.read_csv(params.TLDFile)
+            self.tlds.rename(
+                columns={
+                    params.cat_col: "cat",
+                    params.min_col: "min",
+                    params.max_col: "max",
+                    params.ave_col: "ave",
+                    params.trips_col: "trips",
+                },
+                inplace=True,
+            )
+            self.lookup = pd.read_csv(params.TldLookupFile)
+            self.lookup.rename(
+                columns={params.lookup_zone_col: "zone", params.lookup_cat_col: "cat"},
+                inplace=True,
+            )
+            self.tlds.set_index("cat", inplace=True)
+            self.lookup.sort_values("zone")
+            self.init_params = params.init_params
+            self.dists = self.process_tlds()
+            self.log_path = params.log_path
+            self.furness_tol = params.furness_tolerance
+            self.furness_jac = params.furness_jac
+
+    def process_tlds(self):
+        """Get distributions in the right format for a multi-area gravity model"""
+        dists = []
+        for cat in self.tlds.index.unique():
+            tld = self.tlds.loc[cat]
+            tld = cost_utils.CostDistribution(tld)
+            zones = self.lookup[self.lookup["cat"] == cat].index.values
+            if len(zones) == 0:
+                raise ValueError(
+                    f"{cat} doesn't seem to appear in the given tld "
+                    "lookup. Check for any typos (e.g. lower/upper case). "
+                    f"If this is expected, remove {cat} from your "
+                    "tlds dataframe before inputting."
+                )
+
+            distribution = MultiCostDistribution(
+                name=cat, cost_distribution=tld, zones=zones, function_params=self.init_params
+            )
+            dists.append(distribution)
+
+        return dists
+
+    def _calculate_perceived_factors(
+        self,
+        target_cost_distribution: cost_utils.CostDistribution,
+        achieved_band_shares: np.ndarray,
+    ) -> None:
+        raise NotImplementedError("WIP")
+
+    @property
+    def multi_achieved_band_shares(self) -> np.ndarray:
+        """Analogous to _achieved_band_shares but for a multi-tld"""
+        if self.achieved_cost_dist is None:
+            raise ValueError("Gravity model has not been run. achieved_band_share is not set.")
+        shares = []
+        for dist in self.achieved_cost_dist:
+            shares.append(dist.band_share_vals)
+        return np.concatenate(shares)
+
+    def _create_seed_matrix(self, cost_distributions, cost_args, params_len):
+        base_mat = np.zeros(self.cost_matrix.shape)
```

Comment:
Better off doing this to retain typing etc from base matrix
```suggestion
        base_mat = np.zeros_like(self.cost_matrix)
```